client/src/elements/Prompt.jsx

```diff
@@ -21,33 +32,57 @@ export function Prompt({ file, name, shared }) {
   const progressLabel = player.get("progressLabel");
   const { text: promptString, error: fetchError } = useText({ file });
   const permalink = usePermalink(file);
+
   const [responses, setResponses] = React.useState([]);
 
   if (fetchError) {
     return <p>Error loading prompt, retrying...</p>;
   }
-  if (!promptString) return <p>Loading prompt...</p>;
+
+  if (!promptString) return <p>Prompt file is empty</p>;
 
   // Parse the prompt string into its sections
-  const sectionRegex = /---\n/g;
   const [, metaDataString, prompt, responseString] =
-    promptString.split(sectionRegex);
+    promptString.split(/^-{3,}$/gm);
+
+  // Check if each section is present
+  if (metaDataString === undefined || prompt === undefined || responseString === undefined) {
+    return (
+      <>
+        <p>Error: at least one of metadata, prompt, or response sections are missing. Please verify the formatting of the prompt markdown file.</p>
```

Comment:
What's returned here will display to the user, if there's an error. Hopefully there won't be one in production, but just in case, lets make sure that what gets displayed is ok to show. Why not check for each section independently, and display: "Error: missing prompt metadata", or "Error: missing prompt text". 

Also, how does this handle situations where there is no response string, because it's an openResponse type prompt? Will that return an empty string?

We can put more detailed error messages on the console.log, if that's helpful?